### PR TITLE
chore: refactoring feature flag implementation as per the pr to EE repo

### PR DESCRIPTION
**Pull request by tkAppsmith**
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/24227
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
#### How Has This Been Tested?
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/configurations/FeatureFlagValidationContextConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Setter
+@Configuration
+public class FeatureFlagValidationContextConfig {
+
+    @Autowired
+    public CurrentUserValidationContextProvider currentUserValidationContextProvider;
+
+    @Bean
+    public Map<FeatureFlagEnum, FeatureFlagValidationContextProvider<?>> featureFlagEnumValidationContextMap() {
+        return new EnumMap<>(FeatureFlagEnum.class);
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/configurations/FeatureFlagValidationContextConfig.java:24]: QQ: Where do we intend to initialize values for this?

**Comment by tkAppsmith** [comment]
Closing this PR as the feature is being scraped altogether. The code pattern used to refactor here will be moved to a document for further reference as discussed offline.

**Pull request closed without merging**